File: conformance/bundles/11-emoji-offsets/greet.steps.ts
import { steps } from '@varar/varar'

const { sensor } = steps<Record<string, never>>(() => ({}))

// Two slots: the {string} capture and the trailing (non-header-bound) table.
// Both are echoed back so the core actually compares them — the table's data
// rows only, since the header row is labels and is never compared.
sensor(
  'I greet {string}',
  (_state, name: string, table: ReadonlyArray<ReadonlyArray<string>>) => [name, table.slice(1)],
)

File: conformance/bundles/11-emoji-offsets/greet.steps.rb
require "varar"

steps do
  # Two slots: the {string} capture and the trailing (non-header-bound) table,
  # which arrives in the splat. Both are echoed back so the core actually
  # compares them — the table's data rows only, since the header row is labels.
  sensor("I greet {string}") { |_state, s, *extra| [s, extra[0][1..]] }
end
